jmx-scraper/src/main/resources/solr.yaml

```diff
@@ -0,0 +1,110 @@
+---
+
+rules:
+
+  - bean: solr:dom1=core,dom2=*,category=SEARCHER,scope=searcher,name=numDocs
+    mapping:
+      Value:
+        metric: solr.document.count
+        type: updowncounter
+        unit: "{document}"
+        desc: The total number of indexed documents.
+        metricAttribute:
+          core: param(dom2)
+
+  - bean: solr:dom1=core,dom2=*,category=INDEX,name=sizeInBytes
+    mapping:
+      Value:
+        metric: solr.index.size
+        type: updowncounter
+        unit: By
+        desc: The total index size.
+        metricAttribute:
+          core: param(dom2)
+
+  - beans:
+      - solr:dom1=core,dom2=*,category=QUERY,scope=*,name=requests
+      - solr:dom1=core,dom2=*,category=UPDATE,scope=*,name=requests
+    mapping:
+      Count:
+        metric: solr.request.count
+        type: counter
+        unit: "{query}"
+        desc: The number of queries made.
+        metricAttribute:
+          core: param(dom2)
+          type: param(category)
+          handler: param(scope)
+
+  - beans:
+      - solr:dom1=core,dom2=*,category=QUERY,scope=*,name=requestTimes
+      - solr:dom1=core,dom2=*,category=UPDATE,scope=*,name=requestTimes
+    mapping:
+      Mean:
+        metric: solr.request.time.average
+        type: gauge
+        unit: ms
+        desc: "The average time of a query, based on Solr's histogram configuration."
```

Comment:
nit: I like the word "duration" here in place of "time" -- slightly less room for ambiguity.

Also don't we want to normalize units to seconds?

Reply:
The plan for now is to provide the same metrics as the ones provided by the jmx gatherer, so we keep the current metric names. But however the long term plan is being able to enhance them in the future and remove duplication between instrumentation and jmx-scraper.

For the normalization of value to seconds, while I see the benefit of consistency on the consumer side, it brings a few extra challenges here: we have to add a way to convert existing values (this could for example be done automatically from the provided unit when it's time-related), when capturing very small values like nanoseconds or microseconds it could introduce some challenges with float conversions and lack of precision, and it makes debugging and understanding harder because the value captured at the JMX level is documented in a unit and the produced data would be in another unit which could be misleading.